analyzer/evmtokens/evm_tokens.go

```diff
@@ -102,7 +102,7 @@ func (m Main) processStaleToken(ctx context.Context, batch *storage.QueryBatch,
 		return fmt.Errorf("token address: %w", err)
 	}
 	//nolint:nestif
-	if staleToken.LastDownloadRound == nil {
+	if staleToken.LastDownloadRound == nil || staleToken.Type == nil {
```

Comment:
how can this be nil now? it's an inner JOIN with evm_tokens having token_type NOT NULL

Reply:
oh wait I was looking at the wrong query

Reply:
okay so the .Type can be nil when the evm tokens analysis stale query's left join on the evm_tokens table has no row in the latter. but we currently don't have that situation other than when we've never attempted to scan that token. that is, you've changed the scanner always to create an entry with non-null type 0 (unsupported).

is this in preparation for later having this situation, where staleToken.LastDownloadRound != nil and staleToken.Type == nil?

Reply:
Ha sorry for sending you on a walk through the code. I was even thinking it might be good to call out this change in the PR :)

I added a comment now:
```
	// These two conditions should be equivalent; only a never-before-downloaded token
	// can have a nil type, because downloading a token always sets the type.
	// We check both just in case, because we later dereference the .Type pointer.
```

In the future (once we track native tokens), this will continue to be the case. Native tokens are special and won't have an entry in the `evm_tokens` table.